File: app/src/main/java/com/example/whattodo/adapter/CreditCardAdapter.java
package com.example.whattodo.adapter;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.example.whattodo.CCUtils.CreditCardUtils;

import com.example.whattodo.R;
import com.example.whattodo.model.CreditCard;
import com.firebase.ui.firestore.FirestoreRecyclerAdapter;
import com.firebase.ui.firestore.FirestoreRecyclerOptions;



public class CreditCardAdapter extends FirestoreRecyclerAdapter<CreditCard, CreditCardAdapter.CreditCardHolder> {
    private FirestoreRecyclerOptions<CreditCard> ccList;

    public CreditCardAdapter(@NonNull FirestoreRecyclerOptions<CreditCard> options) {
        super(options);
        this.ccList = options;
    }

    public class CreditCardHolder extends RecyclerView.ViewHolder {
        public ImageView mImageCard;
        public TextView mCardNumberText, mCardDateValidate;
        public CheckBox checkBox;

        public CreditCardHolder (@NonNull View itemView){
            super(itemView);
            mImageCard = itemView.findViewById(R.id.imageCard);
            mCardNumberText = itemView.findViewById(R.id.textCardNumber);
            mCardDateValidate = itemView.findViewById(R.id.textValidateDate);
            checkBox = itemView.findViewById(R.id.checkbox);

            checkBox.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    ccList.getSnapshots().get(getAdapterPosition()).setSelected(checkBox.isChecked());
                }
            });
        }
    }


    @Override
    public CreditCardHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_creditcard, parent, false);
        return new CreditCardHolder(v);
    }


    @Override
    protected void onBindViewHolder(@NonNull CreditCardHolder holder, int position, @NonNull final CreditCard creditCard) {

        CreditCardUtils ccUtils = new CreditCardUtils();
        int cardType = ccUtils.getCardType(creditCard.getNumTargeta());

        if(cardType == 1){
            holder.mImageCard.setImageDrawable(ContextCompat.getDrawable(holder.mImageCard.getContext(), R.drawable.ic_visa));
        }else if(cardType == 2){
            holder.mImageCard.setImageDrawable(ContextCompat.getDrawable(holder.mImageCard.getContext(), R.drawable.ic_mastercard));
        }else if(cardType == 3){
            holder.mImageCard.setImageDrawable(ContextCompat.getDrawable(holder.mImageCard.getContext(), R.drawable.ic_discover));
        }else{
            holder.mImageCard.setImageDrawable(ContextCompat.getDrawable(holder.mImageCard.getContext(), R.drawable.ic_amex));
        }

        holder.mCardNumberText.setText("**** **** **** " +creditCard.getNumTargeta().substring(15,19));
        holder.mCardDateValidate.setText("Expira: "+creditCard.getExpiritDate());

        holder.checkBox.setChecked(false);
    }

    public FirestoreRecyclerOptions<CreditCard> getCreditCards(){
        return ccList;
    }


}

File: app/src/main/java/com/example/whattodo/adapter/CreditCardAdapter.kt
package com.example.whattodo.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.example.whattodo.CCUtils.CreditCardUtils
import com.example.whattodo.R
import com.example.whattodo.adapter.CreditCardAdapter.CreditCardHolder
import com.example.whattodo.model.CreditCard
import com.firebase.ui.firestore.FirestoreRecyclerAdapter
import com.firebase.ui.firestore.FirestoreRecyclerOptions

class CreditCardAdapter(val creditCards: FirestoreRecyclerOptions<CreditCard>) : FirestoreRecyclerAdapter<CreditCard, CreditCardHolder>(creditCards) {

    inner class CreditCardHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mImageCard: ImageView
        var mCardNumberText: TextView
        var mCardDateValidate: TextView
        var checkBox: CheckBox

        init {
            mImageCard = itemView.findViewById(R.id.imageCard)
            mCardNumberText = itemView.findViewById(R.id.textCardNumber)
            mCardDateValidate = itemView.findViewById(R.id.textValidateDate)
            checkBox = itemView.findViewById(R.id.checkbox)
            checkBox.setOnClickListener { creditCards.snapshots[adapterPosition].isSelected = checkBox.isChecked }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CreditCardHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_creditcard, parent, false)
        return CreditCardHolder(v)
    }

    override fun onBindViewHolder(holder: CreditCardHolder, position: Int, creditCard: CreditCard) {
        val cardType = CreditCardUtils.getCardType(creditCard.numTargeta!!)
        if (cardType == 1) {
            holder.mImageCard.setImageDrawable(ContextCompat.getDrawable(holder.mImageCard.context, R.drawable.ic_visa))
        } else if (cardType == 2) {
            holder.mImageCard.setImageDrawable(ContextCompat.getDrawable(holder.mImageCard.context, R.drawable.ic_mastercard))
        } else if (cardType == 3) {
            holder.mImageCard.setImageDrawable(ContextCompat.getDrawable(holder.mImageCard.context, R.drawable.ic_discover))
        } else {
            holder.mImageCard.setImageDrawable(ContextCompat.getDrawable(holder.mImageCard.context, R.drawable.ic_amex))
        }
        holder.mCardNumberText.text = "**** **** **** " + creditCard.numTargeta!!.substring(15, 19)
        holder.mCardDateValidate.text = "Expira: " + creditCard.expiritDate
        holder.checkBox.isChecked = false
    }
}